### PR TITLE
DR-1266: Workflow for running Datarepo integration tests against alpha

### PR DESCRIPTION
**Pull request by choover-broad**
As discussed with @fboulnois, this workflow will be manually triggered by fc-jenkins once all Terra apps, including Data Repo, have been deployed to `alpha`

### CODE REVIEW

**Review by smark88 — APPROVED** [review]

File: .github/workflows/alpha-integration-tests.yaml
Comment (smark88) [inline .github/workflows/alpha-integration-tests.yaml]:
```suggestion
        echo "::set-env name=GOOGLE_APPLICATION_CREDENTIALS::${GOOGLE_APPLICATION_CREDENTIALS}"
         ::add-mask::{GOOGLE_APPLICATION_CREDENTIALS}

```

File: .github/workflows/alpha-integration-tests.yaml
Comment (smark88) [inline .github/workflows/alpha-integration-tests.yaml]:
```suggestion
        echo "::set-env name=GOOGLE_SA_CERT::${GOOGLE_SA_CERT}"
        ::add-mask::{GOOGLE_SA_CERT}
```

File: .github/workflows/alpha-integration-tests.yaml
Comment (fboulnois) [reply .github/workflows/alpha-integration-tests.yaml]: This won't prevent the contents of the file from being printed.

File: src/main/resources/application-alpha.properties
Comment (fboulnois) [inline src/main/resources/application-alpha.properties]: These two variables will live in the alpha helm: https://github.com/broadinstitute/terra-helmfile/pull/156

Comment (snf2ye) [reply src/main/resources/application-alpha.properties]: Should these be moved before merging this?

File: src/test/java/bio/terra/integration/FileTest.java
Comment (snf2ye) [inline src/test/java/bio/terra/integration/FileTest.java:49]: Same with this - we can set the active profiles in alpha helm, right?

File: .github/workflows/alpha-integration-tests.yaml
Comment (snf2ye) [inline .github/workflows/alpha-integration-tests.yaml]: use the PGHOST variable here?

File: src/main/resources/application.properties
Comment (marikomedlock) [inline src/main/resources/application.properties]: This is the name of the SAM resource for Data Repo, which is not necessarily the same as the Google project name -- though they may have the same value in practice. Perhaps we can move this variable to helm, similar to the sam.basePath property above?

File: src/test/java/bio/terra/integration/FileTest.java
Comment (marikomedlock) [inline src/test/java/bio/terra/integration/FileTest.java]: I think this means that the alpha profile will be set to active every time this test is run -- is that what is intended? It seems confusing for when we run this, for example, locally.

Comment (fboulnois) [reply src/test/java/bio/terra/integration/FileTest.java]: The alpha profile won't exist locally, so in practice this is a no-op. Unfortunately `@ActiveProfiles` ignores anything from environment variables and `SPRING_PROFILES_ACTIVE`. There may be a way to further change the integration tests, but I think that is outside the scope of this PR.

File: src/main/resources/application-alpha.properties
Comment (fboulnois) [reply src/main/resources/application-alpha.properties]: I agree that these should be committed in the helm PR and this file removed before we merge.

File: src/test/java/bio/terra/integration/FileTest.java
Comment (fboulnois) [reply src/test/java/bio/terra/integration/FileTest.java:49]: We set `SPRING_PROFILES_ACTIVE` to `alpha` in Helm, but the special `@ActiveProfiles` annotation ignores anything from environment variables and `SPRING_PROFILES_ACTIVE`, so we still need it for the integration tests.

File: .github/workflows/alpha-integration-tests.yaml
Comment (snf2ye) [inline .github/workflows/alpha-integration-tests.yaml]: After looking at the helm definitions, I think we could actually remove any env variable that's already defined there. 
* remove SPRING_PROFILES_ACTIVE, GOOGLE_CLOUD_PROJECT
* We could remove GOOGLE_CLOUD_DATA_PROJECT and reference [GOOGLE_SINGLEDATAPROJECTID](https://github.com/broadinstitute/terra-helmfile/pull/156/files#diff-9315f4b742c97ef06713007f74015e07L48) instead below. But, we're going to have to change how we reference data projects before "going live" so it might be good to have it defined here. 
* IT_JADE_API_URL could be added to helm & removed here, [like we're doing in some of the other environments](https://github.com/broadinstitute/datarepo-helm-definitions/blob/master/dev/dev/devDeployment.yaml#L33)

File: src/main/resources/application.properties
Comment (fboulnois) [reply src/main/resources/application.properties]: After discussion with Mariko, here is the suggested path forward:

1. Short term, I will revert this change and move the variable to helm.
2. Long term, the team will discuss standardizing it across environments and setting a default in `application.properties`
3. Helm file overrides will be created in each environment as required

File: src/test/java/bio/terra/integration/FileTest.java
Comment (marikomedlock) [reply src/test/java/bio/terra/integration/FileTest.java]: Okay sorry for being dense, but then I'm confused what the purpose of the profile is. If it's a no-op and not defined anywhere, then what effect does it have on the test?

File: .github/workflows/alpha-integration-tests.yaml
Comment (fboulnois) [reply .github/workflows/alpha-integration-tests.yaml]: 1. `SPRING_PROFILES_ACTIVE` is required so that `@ActiveProfiles` actually uses that annotation and the `alpha` overrides
2. I may be able to remove `GOOGLE_CLOUD_PROJECT`, but I think it is needed for `gcloud` commands
3. `GOOGLE_CLOUD_DATA_PROJECT` is used for `get-iam-policy`
4. `IT_JADE_API_URL` needs to be defined locally on the GitHub Runner for integration tests in addition to remotely, otherwise tests will error.

File: src/main/resources/application-alpha.properties
Comment (ddietterich) [reply src/main/resources/application-alpha.properties]: So we will not need to have an alpha profile, right?

File: src/test/java/bio/terra/integration/FileTest.java
Comment (ddietterich) [reply src/test/java/bio/terra/integration/FileTest.java]: I think we should not include this and use environment variables for the different test environments.

**Review by ddietterich — CHANGES_REQUESTED** [review]
Alpha profile questions

File: src/test/java/bio/terra/integration/FileTest.java
Comment (fboulnois) [reply src/test/java/bio/terra/integration/FileTest.java]: It's defined in the datarepo [alpha.yaml](https://github.com/broadinstitute/terra-helmfile/blob/master/terra/values/datarepo/live/alpha.yaml#L50) .

**Review by snf2ye — APPROVED** [review]

**Review by ddietterich — APPROVED** [review]

File: .github/workflows/alpha-integration-tests.yaml
Comment (snf2ye) [inline .github/workflows/alpha-integration-tests.yaml:4]: If you changes these to just "jade-dev-account.json" & "jade-dev-account.pem", then should only need to write the vault tokens once.  (this is the finding from working on the int test run -> the /tmp folder isn't mounted, so when docker power cycles on each step, the credentials are lost)

File: .github/workflows/alpha-integration-tests.yaml
Comment (snf2ye) [inline .github/workflows/alpha-integration-tests.yaml:38]: Are these used anywhere?

Comment (fboulnois) [inline .github/workflows/alpha-integration-tests.yaml:38]: Yes, these are used to cleanup the IAM policies using the Alpha service account.

File: .github/workflows/alpha-integration-tests.yaml
Comment (fboulnois) [reply .github/workflows/alpha-integration-tests.yaml:4]: The vault tokens are written twice to the same location because one activation is to get the alpha SA for cleaning up the IAM policies, whereas the other activation is to run integration with the dev SA.

Comment (fboulnois) [reply .github/workflows/alpha-integration-tests.yaml:4]: None of the action is running in containers so `/tmp` is not an issue.

**Review by snf2ye — APPROVED** [review]

**Review by smark88 — APPROVED** [review]
Approved pending this run i kicked off otherwise the code looks good so far

https://github.com/DataBiosphere/jade-data-repo/runs/1102760669?check_suite_focus=true

**Pull request merged**
